jsonrpc/handlerT.go

```diff
@@ -108,3 +114,47 @@ func (h *HandlerT) StopGoTrace() error {
 
 	return nil
 }
+
+// Stacks returns a printed representation of the stacks of all goroutines. It
+// also permits the following optional filters to be used:
+//   - filter: boolean expression of packages to filter for
+func (*HandlerT) Stacks(filter *string) (string, error) {
+	buf := new(bytes.Buffer)
+	err := pprof.Lookup("goroutine").WriteTo(buf, 2)
+
+	if err != nil {
+		return "", err
+	}
+
+	// If any filtering was requested, execute them now
+	if filter != nil && len(*filter) > 0 {
+		expanded := *filter
+
+		// The input filter is a logical expression of package names. Transform
+		// it into a proper boolean expression that can be fed into a parser and
+		// interpreter:
+		//
+		// E.g. (eth || snap) && !p2p -> (eth in Value || snap in Value) && p2p not in Value
+		expanded = regexp.MustCompile(`[:/\.A-Za-z0-9_-]+`).ReplaceAllString(expanded, "`$0` in Value")
+		expanded = regexp.MustCompile("!(`[:/\\.A-Za-z0-9_-]+`)").ReplaceAllString(expanded, "$1 not")
```

Comment:
In addition to Goran's comment I would consider changing single quotes to double quotes. When we do so the 1st MustCompile expression will be red because it differs from the 2nd (which seems to be correct) because it has missing \ before .A
Beside that I don't know do we really need single quote around $0? Below is an example that hopefully would work, but it needs to be tested.

expanded = regexp.MustCompile("[:/\\.A-Za-z0-9_-]+").ReplaceAllString(expanded, "$0 in Value")
expanded = regexp.MustCompile("!('[:/\\.A-Za-z0-9_-]+')").ReplaceAllString(expanded, "$1 not")

Reply:
I would say it works like this:

videntifierRegex := regexp.MustCompile(`[:/\.A-Za-z0-9_-]+`)
notRegex := regexp.MustCompile(`!([:/\.A-Za-z0-9_-]+)`)

expanded := identifierRegex.ReplaceAllString(filter, "`$0` in Value")
expanded = notRegex.ReplaceAllString(expanded, "`$1` not")

Reply:
I would say it works like this:

identifierRegex              = regexp.MustCompile(`[:/\.A-Za-z0-9_-]+`)
notRegex                     = regexp.MustCompile(`!([:/\.A-Za-z0-9_-]+)`)

expanded := identifierRegex.ReplaceAllString(filter, "`$0` in Value")
expanded = notRegex.ReplaceAllString(expanded, "`$1` not")